Attribute #[clearbody] useful instead of Let ... Qed (#17544)
coq/coq: doc/sphinx/language/core/sections.rst

   In this case, the proof should be terminated with :cmd:`Defined` in order to define a constant
   for which the computational behavior is relevant.  See :ref:`proof-editing-mode`.

.. attr:: clearbody

>    Using this attribute with a :cmd:`Let` in a section will cause the
>    body of the definition to be cleared in the proof context of following proofs.

Reviewer: ```suggestion
   When used with :cmd:`Let` in a section,
   clears the body of the definition in the proof context of following proofs.
```

.. warn:: @ident is declared opaque but this is not fully respected inside the section and not at all outside the section.

>    Terminating the proof for a :cmd:`Let` with :cmd:`Qed` is deprecated.

Reviewer: ```suggestion
   Terminating the proof for a :cmd:`Let` with :cmd:`Qed` is deprecated; the proof
   should be terminated with :cmd:`Defined`.  Use :attr:`clearbody` with :cmd:`Defined`
   to clear the body of the proof when the section is closed.
```

and adjust the following lines (77-82).

Also, would you move the part about "The variable will appear to have no body inside proofs but the kernel will still use the body when checking" into the description of `clearbody`" if it is specific to `clearbody`?  If it's standard behavior for `Qed` (just opaqueness?) we shouldn't mention that important fact only in the description of a warning.  If it's just opaqueness, maybe use that term in the description.

   The variable will appear to have no body inside proofs but the kernel will still use
   the body when checking.  When the section is closed it will be treated the as if
   :cmd:`Defined` had been specified.

   This is the same behavior as :attr:`clearbody` so if it is what you

>    intend use the attribute instead.

Reviewer 2: Then, should we go beyond warning, and actually deprecate `Let ... Qed.` in favor of `#[ clearbody ] Let ... Defined.`?
